src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -70,28 +70,52 @@ private async Task<bool> IsEnabledAsync<TContext>(string feature, TContext appCo
                 }
             }
 
-            bool enabled = false;
+            bool enabled;
 
             FeatureDefinition featureDefinition = await _featureDefinitionProvider.GetFeatureDefinitionAsync(feature).ConfigureAwait(false);
 
             if (featureDefinition != null)
             {
-                //
-                // Check if feature is always on
-                // If it is, result is true, goto: cache
+                if (featureDefinition.RequirementType == RequirementType.All && _options.IgnoreMissingFeatureFilters)
+                {
+                    throw new FeatureManagementException(
+                        FeatureManagementError.Conflict, 
+                        $"The 'IgnoreMissingFeatureFilters' flag cannot use used in combination with a feature of requirement type 'All'.");
+                }
 
-                if (featureDefinition.EnabledFor.Any(featureFilter => string.Equals(featureFilter.Name, "AlwaysOn", StringComparison.OrdinalIgnoreCase)))
+                //
+                // Treat an empty list of enabled filters as a disabled feature
+                if (featureDefinition.EnabledFor == null || featureDefinition.EnabledFor.Count() == 0)
```

Comment:
`featureDefinition.EnabledFor.Count() == 0` I'd recommend `featureDefinition.EnabledFor.Any()`. Otherwise you may need to enumerate an entire enumerable to get the entire count.